### PR TITLE
Upload unfinished job/step logs from runner on cancellation timeout

### PR DESCRIPTION
**Pull request by TingluoHuang**
#254

### CODE REVIEW

File: src/Runner.Common/Logging.cs [hunk 1]
```diff
@@ -40,7 +39,6 @@ public override void Initialize(IHostContext hostContext)
         {
             base.Initialize(hostContext);
             _totalLines = 0;
-            _pageId = Guid.NewGuid().ToString();
```
Comment (TingluoHuang) [inline src/Runner.Common/Logging.cs]: we don't really need a new guid for each page, we can leverage the timeline record id for each steps, those are already unique per step.

File: src/Runner.Listener/JobDispatcher.cs [hunk 2]
```diff
@@ -712,6 +716,116 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
...
+
+            var logs = Directory.GetFiles(logFolder);
+            if (logs.Length == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection));
```
Comment (TingluoHuang) [inline src/Runner.Listener/JobDispatcher.cs]: pull the cred from the job message, we will use this cred to talk to actions service and upload the log.

File: src/Runner.Listener/JobDispatcher.cs [hunk 3]
```diff
@@ -712,6 +716,116 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
...
+
+                        if (!int.TryParse(logName.Substring(logPageSeperator + 1), out pageNumber))
+                        {
+                            Trace.Warning($"log file '{log}' doesn't follow naming convension 'GUID_INT'.");
+                            continue;
+                        }
+                    }
+
+                    var record = timeline.Records.FirstOrDefault(x => x.Id == logRecordId);
+                    if (record != null)
```
Comment (TingluoHuang) [inline src/Runner.Listener/JobDispatcher.cs]: We only upload log files to timeline records for this job.

File: src/Runner.Listener/JobDispatcher.cs [hunk 4]
```diff
@@ -712,6 +716,116 @@ public async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockToke
...
+            var logs = Directory.GetFiles(logFolder);
+            if (logs.Length == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                var systemConnection = message.Resources.Endpoints.SingleOrDefault(x => string.Equals(x.Name, WellKnownServiceEndpointNames.SystemVssConnection));
+                ArgUtil.NotNull(systemConnection, nameof(systemConnection));
```
Comment (thboop) [inline src/Runner.Listener/JobDispatcher.cs]: Should we throw here, or should we just return because its best effort?

**Pull request merged**